Makefile

```diff
@@ -104,7 +104,7 @@ test: manifests generate fmt vet test-unit test-e2e #HELP Run all tests.
 
 .PHONY: e2e
 FOCUS := $(if $(TEST),-v -focus "$(TEST)")
-E2E_FLAGS ?= ""
+E2E_FLAGS ?= --label-filter="!ForceSemverUpgradeConstraints=false"
```

Comment:
Created #447 for better switching between feature gate states.

For now: we have E2E tests for both true and false values. But for now we will be excluding tests for `ForceSemverUpgradeConstraints=false` feature gate state from run since default is `ForceSemverUpgradeConstraints=true`.

Reply:
Unless it's somehow very challenging, IMHO makes a lot of sense to only check in tests that run - is it a big lift to get working for this pR?

Reply:
> Unless it's somehow very challenging, IMHO makes a lot of sense to only check in tests that run

Just to clarify: tests we no longer run in this PR exist in `main`. What I did:
* Moved existing tests around
* Labeled them
* Excluded from execution since default behaviour changed.

Didn't want to remove existing tests because I hope that we will eventually run them (#447).

> is it a big lift to get working for this pR?

I would like to keep this PR focused on semver and leave E2E.

But I'm happy to look into this as a separate effort. If you have any suggestions how to do it - could you please point into the right direction on #447?

Reply:
What about using unit tests of the Reconcile function to test both feature gate states?

Then the e2e can simply test the default state of the feature gate.

Reply:
As an aside: I have a pretty strong opinion that we should test as much as possible in `Reconcile` tests so that our e2e's remain light and fast and relatively resistant to change.

Reply:
> What about using [unit tests of the Reconcile function](https://github.com/operator-framework/operator-controller/blob/main/internal/controllers/operator_controller_test.go) to test both feature gate states?
> 
> Then the e2e can simply test the default state of the feature gate.

We do that on the variable source level already [here](https://github.com/operator-framework/operator-controller/pull/444/files#diff-d617afce32874544d5987c6a05c4b55525d66d1150c18330e9e70352d7959698R125) (there is another test in this file for disabled state which is no in the diff).

Reply:
My point is: rather than continuing to add more e2e tests, can we instead add more controller reconcile tests?

Is there a reason we think what's being tested in the new e2es from this PR can't instead be tested in controller reconcile tests?

Incidentally, it would solve this problem with inert tests because we can really easily enable/disable the gate in those tests.